src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -22,108 +23,161 @@ class AblyBroadcaster extends Broadcaster
+     * @param $ably \Ably\AblyRest
      * @return void
      */
-    public function __construct(AblyRest $ably)
+    public function __construct(AblyRest $ably, $config)
     {
         $this->ably = $ably;
+        if (self::$serverTimeDiff == null) {
+            self::setServerTime(round($this->ably->time() / 1000));
+        }
+        if (array_key_exists('disable_public_channels', $config) && $config['disable_public_channels']) {
+            $this->defaultChannelClaims = ['public:*' => ['channel-metadata']];
+        }
+        if (array_key_exists('token_expiry', $config)) {
+            $this->tokenExpiry = $config['token_expiry'];
+        }
     }
 
+    private static $serverTimeDiff = null;
+
     /**
-     * Authenticate the incoming request for a given channel.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return mixed
-     *
-     * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+     * @param $time int
+     * @return void
      */
-    public function auth($request)
+    private static function setServerTime($time)
     {
-        $channelName = $this->normalizeChannelName($request->channel_name);
+        self::$serverTimeDiff = time() - $time;
+    }
 
-        if (empty($request->channel_name) ||
-            ($this->isGuardedChannel($request->channel_name) &&
-            ! $this->retrieveUser($request, $channelName))) {
-            throw new AccessDeniedHttpException;
+    /**
+     * @return int
+     */
+    private static function getServerTime()
+    {
+        if (self::$serverTimeDiff != null) {
+            return time() - self::$serverTimeDiff;
         }
 
-        return parent::verifyUserCanAccessChannel(
-            $request, $channelName
-        );
+        return time();
     }
 
     /**
-     * Return the valid authentication response.
+     * Get the public token value from the Ably key.
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  mixed  $result
      * @return mixed
      */
-    public function validAuthenticationResponse($request, $result)
+    protected function getPublicToken()
     {
-        if (str_starts_with($request->channel_name, 'private')) {
-            $signature = $this->generateAblySignature(
-                $request->channel_name, $request->socket_id
-            );
-
-            return ['auth' => $this->getPublicToken().':'.$signature];
-        }
+        return Str::before($this->ably->options->key, ':');
+    }
 
-        $channelName = $this->normalizeChannelName($request->channel_name);
+    /**
+     * Get the private token value from the Ably key.
+     *
+     * @return mixed
+     */
+    protected function getPrivateToken()
+    {
+        return Str::after($this->ably->options->key, ':');
+    }
 
-        $user = $this->retrieveUser($request, $channelName);
+    /**
+     * Authenticate the incoming request for a given channel.
+     *
+     * @param $request \Illuminate\Http\Request
```

Comment:
```suggestion
     * @param  \Illuminate\Http\Request  $request
```
First variable type and then the variable name

Reply:
Fixed!